spinoso-time/src/time/tzrs/offset.rs

```diff
@@ -64,15 +65,14 @@ fn local_time_zone() -> TimeZoneRef<'static> {
 /// Generates a [+/-]HHMM timezone format from a given number of seconds
 /// Note: the actual seconds element is effectively ignored here
 #[inline]
-#[must_use]
-fn offset_hhmm_from_seconds(seconds: i32) -> String {
+fn offset_hhmm_from_seconds(seconds: i32, buf: &mut [u8; 5]) -> io::Result<()> {
     let flag = if seconds < 0 { '-' } else { '+' };
     let minutes = seconds.abs() / 60;
 
     let offset_hours = minutes / 60;
     let offset_minutes = minutes - (offset_hours * 60);
 
-    format!("{}{:0>2}{:0>2}", flag, offset_hours, offset_minutes)
+    write!(buf.as_mut_slice(), "{flag}{offset_hours:0>2}{offset_minutes:0>2}")
```

Comment:
Ahhhh, this is pretty cool. I see why you're writing to the passed in buffer vs. attempting to return a `[u8;5]`. I assume the `io::Result` would give an error on overflow of the buffer, so that's how you're able to safely write to the 5 byte buffer sort of thing. Nice.